services/worker-manager/src/providers/aws.js

```diff
@@ -82,83 +64,108 @@ class AwsProvider extends Provider {
       });
     }
 
-    // Make sure we don't get "The same resource type may not be specified more than once in tag specifications" errors
-    const TagSpecifications = config.launchConfig.TagSpecifications ? config.launchConfig.TagSpecifications : [];
-    const instanceTags = [];
-    const otherTagSpecs = [];
-    TagSpecifications.forEach(ts =>
-      ts.ResourceType === 'instance' ? instanceTags.concat(ts.Tags) : otherTagSpecs.push(ts)
-    );
+    const uniqueRegions = new Set(workerPool.config.launchConfigs.map(lc => lc.region));
+    let ec2s = {};
```

Comment:
why don't we make this in the constructor so that they can be cached across iteration loops? And then every time we get a new region we can try to grab the old one or make a new one if we haven't seen it before. No need to make `uniqueRegions` at that point